usd_qtpy/render_util/dialog.py

```diff
@@ -0,0 +1,889 @@
+# Dialog functions and wrappers to do with the rendering API.
+
+import os
+import re
+from typing import Any
+from functools import partial
+
+from qtpy import QtWidgets, QtCore
+from pxr import Usd, UsdGeom, Sdf
+from pxr.Usdviewq.stageView import StageView
+
+from . import playblast, framing_camera, turntable
+from .basecls import RenderReportable
+from ..resources import get_icon
+from ..lib import usd
+
+
+def _rectify_path_framenumberspec(path: str, padding: int =  4):
+    """Ensure a placeholder for framenumbers exists in the path.
+
+    >>> _rectify_path_framenumberspec("filename##.png")
+    "filename####.png"
+    >>> _rectify_path_framenumberspec("filename###.png")
+    "filename####.png"
+
+    Returns:
+        str: The path with an ensured frame number hash.
+
+    """
+    base, file = os.path.split(path)
+    file, extension = os.path.splitext(file)
+    # if too little hashes were present, remove them.
+    file = file.replace("#", "")
+    file += "#" * padding
+
+    return os.path.join(base, file + extension)
+
+
+def prompt_input_path(caption: str="Load item") -> str:
+    
+    filename, _ = QtWidgets.QFileDialog.getOpenFileName(
+        caption=caption,
+        filter="USD (*.usd *.usda *.usdc)"
+    )
+    return filename
+
+
+def prompt_output_path(caption="Save frame"):
+    """Prompt for render filename, ensure frame hashes are present in path"""
+    filename, _ = QtWidgets.QFileDialog.getSaveFileName(
+        caption=caption,
+        filter="PNG (*.png);;JPEG (*.jpg, *.jpeg)"
+    )
+
+    # ensure there's at least two or more hashes, if not, put in 4.
+    REGEX_HASH = r"(#{2,})"
+
+    hash_match = re.search(REGEX_HASH, filename)
+    if not hash_match:
+        filename = _rectify_path_framenumberspec(filename)
+
+    return filename
```

Comment:
This should return early if no filename picked, otherwise canceling the prompt results in `####`

Also, I'd prefer it if picking e.g. `turntable.0001.png` doesn't result in `turntable.1001####.png` but instead: `turntable.####.png`.

Reply:
That can be done!
I'll implement this asap.

Reply:
![image](https://github.com/BigRoy/usd-qtpy/assets/87427129/0b99abd1-ea73-4995-87f4-169d8a092f19)